From 146395e3dac0015850d0fd61f9db206499161e1c Mon Sep 17 00:00:00 2001
From: Peter Volkov <pva@gentoo.org>
Date: Mon, 20 Jun 2011 10:50:29 +0400
Subject: [PATCH] build: use absolute path for M=

Use absolute path for M during checking kernelrelease. This will force
temporary objects be built in the current directory and not $kbuilddir
as it happenes with the current kernel scripts/Kbuild.include, where
e.g. try-run target called with TMPOUT=M=. while at the time it is
called pwd=$kbuilddir. This should fix sandbox violation (modification
of files outside build dir) in Gentoo:
https://bugs.gentoo.org/show_bug.cgi?id=371997
---
 configure.ac |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index 0660eea..6361e48 100644
--- a/configure.ac
+++ b/configure.ac
@@ -42,7 +42,7 @@ regular_CFLAGS="-Wall -Waggregate-return -Wmissing-declarations \
 
 if test -n "$kbuilddir"; then
 	AC_MSG_CHECKING([kernel version that we will build against])
-	krel="$(make -sC "$kbuilddir" M=. kernelrelease)";
+	krel="$(make -sC "$kbuilddir" M="$(pwd)" kernelrelease)";
 	kmajor="${krel%%[[^0-9]]*}";
 	kmajor="$(($kmajor+0))";
 	krel="${krel:${#kmajor}}";
-- 
1.7.3.4

